drake/systems/DrakeSystem.cpp

```diff
@@ -33,21 +36,59 @@ DrakeSystem::DrakeSystem(const std::string &_name, unsigned int num_continuous_s
           state_frame(make_shared<MultiCoordinateFrame>(name+"State",initializer_list<CoordinateFramePtr>({continuous_state_frame,discrete_state_frame}))),
           output_frame(make_shared<CoordinateFrame>(name+"Output",num_outputs,"y")) { }
 
-DrakeSystem::VectorXs DrakeSystem::getRandomState() {
-  return DrakeSystem::VectorXs::Random(state_frame->getDim());
+VectorXd DrakeSystem::getRandomState() {
+  return VectorXd::Random(state_frame->getDim());
 }
 
-DrakeSystem::VectorXs DrakeSystem::getInitialState() {
+VectorXd DrakeSystem::getInitialState() {
   return getRandomState();
 }
 
-void DrakeSystem::simulate(double t0, double tf, const DrakeSystem::VectorXs &x0, const SimulationOptions& options) const {
+DrakeSystemPtr DrakeSystem::tilqr(const Eigen::VectorXd& x0, const Eigen::VectorXd& u0, const Eigen::MatrixXd& Q, const Eigen::MatrixXd& R) {
+  if (!isTimeInvariant())
+  throw runtime_error("DrakeSystem::TILQR.  Time-invariant LQR only makes sense for time invariant systems.");
+
+  if (discrete_state_frame->getDim()>0)
+  throw runtime_error("DrakeSystem::TILQR.  Discrete-time LQR is not implemented yet (but would be easy).");
+  if (continuous_state_frame->getDim()<1)
+  throw runtime_error("DrakeSystem::TILQR.  This system has no continuous states.");
+
+  int num_x = continuous_state_frame->getDim(),
+      num_u = input_frame->getDim();
+
+  Eigen::VectorXd xu(num_x+num_u);
+  xu << x0, u0;
+  Drake::TaylorVecX xu_taylor = Drake::initTaylorVecX(xu);
+
+  auto xdot = autoDiffToGradientMatrix(dynamics(0*xu_taylor(1),xu_taylor.head(num_x),xu_taylor.tail(num_u)));
```

Comment:
did the "just pass double 0" first, but then it increases the number of template instantiations that I needed to support in the base class (because for time-varying systems, we DO need taylorvars here).  will leave it for now, but address it in the newest template parameterization idea.